custom_components/waste_collection_schedule/waste_collection_schedule/source/ccc_tas_gov_au.py

```diff
@@ -0,0 +1,164 @@
+    frequency_recycling = 0
+    next_recycling = date(1970,1,1)
+    frequency_greenwaste = 0
+    next_greenwaste = date(1970,1,1)
+
+    for result in results:
+        if result['name'] == "Waste Collection Day":
+            collection_day = result['value']
+        
+        if result['name'] == 'Garbage Pickup':
+            frequency_rubbish = _calculate_frequency(result['value'])
+            next_rubbish = _next_date(result['value'])
+
+        if result['name'] == 'Recycling Pickup':
+            frequency_recycling = _calculate_frequency(result['value'])
+            next_recycling = _next_date(result['value'])
+
+        if result['name'] == 'Green Waste Pickup':
+            frequency_greenwaste = _calculate_frequency(result['value'])
+            next_greenwaste = _next_date(result['value'])
+
+    rubbish_collection_dates = _get_next_n_dates(
+        next_rubbish, 52, timedelta(days=frequency_rubbish)
+    )
+
+    recycling_collection_dates = _get_next_n_dates(
+        next_recycling, 52, timedelta(days=frequency_recycling)
+    )
+
+    greenwaste_collection_dates = _get_next_n_dates(
+        next_greenwaste, 52, timedelta(days=frequency_greenwaste)
+    )
+
+    entries.extend(
+        [
+            Collection(date=collection_date, t="Rubbish", icon=ICON_MAP["RUBBISH"])
+            for collection_date in rubbish_collection_dates
+        ]
+    )
+
+    entries.extend(
+        [
+            Collection(date=collection_date, t="Recycling", icon=ICON_MAP["RECYCLE"])
+            for collection_date in recycling_collection_dates
+        ]
+    )
+
+    entries.extend(
+        [
+            Collection(date=collection_date, t="Greenwaste", icon=ICON_MAP["ORGANIC"])
+            for collection_date in greenwaste_collection_dates
+        ]
+    )
+    
+    return entries
+
+
+class Source:
+    def __init__(self, address: str):
+        self.address = address
+
+    def fetch(self) -> list[Collection]:
+        session = requests.Session()
+        session.headers.update({"content-type": "application/json"})
+
+        address_search_data = {'address': self.address}
+        address_url = "%s/address-search" % URL
+        address = session.post(address_url, data=json.dumps(address_search_data))
+
+        if address.status_code != 200:
+            raise Exception("Could not complete addres lookup %i" % address.status_code)
+    
+        address_result = json.loads(address.content)
+
+        if not address_result['success']:
+            raise Exception("Could not find address")
+
+        ccc_formatted_address = address_result['results'][0]
+
+        collection_search_data = {'address': ccc_formatted_address}
+        collection_url = "%s/collection-search" % URL
+        collections = session.post(collection_url, data=json.dumps(collection_search_data))
+
+        if address.status_code != 200:
+            raise Exception("Could not find collections")
+
+        collections_result = json.loads(collections.content)
```

Comment:
You can simply this by not using `json.` but the requests built in JSON parsing:

```suggestion
        address = session.post(address_url, json=address_search_data)

        if address.status_code != 200:
            raise Exception("Could not complete addres lookup %i" % address.status_code)
    
        address_result = address.json()

        if not address_result['success']:
            raise Exception("Could not find address")

        ccc_formatted_address = address_result['results'][0]

        collection_search_data = {'address': ccc_formatted_address}
        collection_url = "%s/collection-search" % URL
        collections = session.post(collection_url, json=collection_search_data)

        if address.status_code != 200:
            raise Exception("Could not find collections")

        collections_result = collections.json()
```